src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/listeners/court/libra/LibraCourtCaseListenerIntTest.kt

```diff
@@ -0,0 +1,147 @@
+package uk.gov.justice.digital.hmpps.personrecord.message.listeners.court.libra
+
+import org.assertj.core.api.Assertions.assertThat
+import org.awaitility.kotlin.await
+import org.awaitility.kotlin.matches
+import org.awaitility.kotlin.untilAsserted
+import org.awaitility.kotlin.untilCallTo
+import org.awaitility.kotlin.untilNotNull
+import org.jmock.lib.concurrent.Blitzer
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import software.amazon.awssdk.services.sns.model.MessageAttributeValue
+import software.amazon.awssdk.services.sns.model.PublishRequest
+import uk.gov.justice.digital.hmpps.personrecord.client.model.hmcts.MessageType.LIBRA_COURT_CASE
+import uk.gov.justice.digital.hmpps.personrecord.integration.MessagingMultiNodeTestBase
+import uk.gov.justice.digital.hmpps.personrecord.jpa.entity.PersonEntity
+import uk.gov.justice.digital.hmpps.personrecord.model.identifiers.CROIdentifier
+import uk.gov.justice.digital.hmpps.personrecord.model.identifiers.PNCIdentifier
+import uk.gov.justice.digital.hmpps.personrecord.model.person.Address
+import uk.gov.justice.digital.hmpps.personrecord.model.person.Person
+import uk.gov.justice.digital.hmpps.personrecord.model.types.SourceSystemType
+import uk.gov.justice.digital.hmpps.personrecord.service.type.TelemetryEventType
+import uk.gov.justice.digital.hmpps.personrecord.service.type.TelemetryEventType.CPR_RECORD_CREATED
+import uk.gov.justice.digital.hmpps.personrecord.service.type.TelemetryEventType.HMCTS_MESSAGE_RECEIVED
+import uk.gov.justice.digital.hmpps.personrecord.test.messages.libraHearing
+import uk.gov.justice.hmpps.sqs.countMessagesOnQueue
+import java.time.LocalDate
+import java.util.concurrent.TimeUnit.SECONDS
+
+class LibraCourtCaseListenerIntTest : MessagingMultiNodeTestBase() {
+
+  @BeforeEach
+  override fun beforeEach() {
+    super.beforeEach()
+    personRepository.deleteAll()
+    telemetryRepository.deleteAll()
```

Comment:
Even though we are leaning toward the 'non delete way of test' this seems like the path of less resistance / excessive development time after some trying to locate and verify the record creation on a record with no id's.